### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by apaszke**
* Handle unsqueezed dimensions in Repeat backward (#2095)
* `parallel_apply` can now accept arbitrary inputs (#1992)
* Keyword args are now allows in options that also use `long_args` (#1616)
* `Variable.expand` now accepts all inputs that `Tensor.expand` accepts (#1910)
* Unified argument names for Variable and Tensor methods (#1924, #1839)
* Changed assertions with side effects to `if` statements (#1848)

### CODE REVIEW

File: torch/autograd/variable.py [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 import torch
 import torch._C as _C
+import collections
 from collections import OrderedDict
 import torch.sparse as sparse
 import torch.utils.hooks as hooks
 import warnings
 
+if sys.version_info[0] == 2:
+    Iterable = collections.Iterable
```
Comment (soumith) [inline torch/autograd/variable.py]: Iterable is not used anywhere in this patch

File: torch/autograd/__init__.py [hunk 2]
```diff
@@ -152,5 +152,5 @@ def grad(outputs, inputs, grad_outputs=None, retain_graph=None, create_graph=Non
         outputs, grad_outputs, retain_graph,
         inputs, only_inputs)
 
-status = torch._C._autograd_init()
-assert status, "Autograd failed to initialize."
+if not torch._C._autograd_init():
```
Comment (albanD) [inline torch/autograd/__init__.py]: I thought this was ok this way as there are many places in the code where if an assert is not respected, the rest of the code will fail. Isn't the point of the optimized version to actually disable all these check assuming they will not fail.

Comment (apaszke) [inline torch/autograd/__init__.py]: It is, but this statement is executed only once, so and the overhead of this single boolean comparison is completely irrelevant

File: torch/autograd/_functions/tensor.py [hunk 3]
```diff
@@ -105,12 +105,15 @@ def backward(ctx, grad_output):
 class Expand(Function):
 
     @staticmethod
+    # NOTE: new_size can be a tuple of any arguments that expand accepts, including a single-element
```
Comment (soumith) [inline torch/autograd/_functions/tensor.py]: this commit definitely needs tests

**Pull request merged**